### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by VOID404**
**Description**

```
client -> gateway -> service
       ^A         ^B
client <- gateway <- service
       ^C         ^D
```

We already had logging of `A` and `C` (in some form).
Logging of `B` and `D` would be helpful.

Changes proposed in this pull request:

- Add logging for `B`
- Change (and clean up) logging system

**Related issue(s)**

#17905
#17265

### CODE REVIEW

File: components/central-application-gateway/pkg/httptools/logging.go [hunk 1]
```diff
@@ -1,28 +1,75 @@
...
 		fullPath := r.RequestURI
 		if fullPath == "" {
 			fullPath = r.URL.RequestURI()
 		}
-		proto := r.Proto
 		responseCode := lw.status
 		duration := time.Since(lw.start).Nanoseconds() / int64(time.Millisecond)
 
-		log.Infof("%s: %s %s %s %d %d", label, method, fullPath, proto, responseCode, duration)
+		log := zap.L().Sugar().
```
Comment (mvshao) [inline components/central-application-gateway/pkg/httptools/logging.go:65]: Suggestion: can we also log the`fullPath` variable in default logger? Or if you see a reason not to log this variable by default could you delete the `fullPath` variable as it is not used

File: components/central-application-gateway/cmd/applicationgateway/options.go [hunk 2]
```diff
@@ -2,53 +2,54 @@ package main
...
+	flag.StringVar(&opts.kubeConfig, "kubeConfig", "", "Path to a kubeconfig. Only required if out-of-cluster")
+	opts.logLevel = zap.LevelFlag("logLevel", zap.InfoLevel, "Log level: panic | fatal | error | warn | info | debug. Can't be lower than info")
+	flag.IntVar(&opts.proxyCacheTTL, "proxyCacheTTL", 120, "TTL, in seconds, for proxy cache of Remote API information")
+	flag.IntVar(&opts.proxyPort, "proxyPort", 8080, "Port that acts as a proxy for the calls from services and Functions to an external solution in the default standalone mode or Compass bundles with a single API definition")
+	flag.IntVar(&opts.proxyPortCompass, "proxyPortCompass", 8082, "Port that acts as a proxy for the calls from services and Functions to an external solution in the Compass mode")
+	flag.IntVar(&opts.proxyTimeout, "proxyTimeout", 10, "Timeout for requests sent through the proxy, expressed in seconds")
+	flag.IntVar(&opts.requestTimeout, "requestTimeout", 1, "Timeout for requests sent through Central Application Gateway, expressed in seconds")
```
Comment (mvshao) [inline components/central-application-gateway/cmd/applicationgateway/options.go:33]: Could you unify the timeouts? The default value for `requestTimeout` in Helm values is set to 10 seconds.

File: tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/proxy-cases.yml [hunk 3]
```diff
@@ -36,3 +36,21 @@ spec:
         - type: API
           targetUrl: "http://{{ .Values.global.mockServiceName }}.{{ .Values.global.namespace  }}.svc.cluster.local:8080/v1/api/unsecure/code/203"
           centralGatewayUrl: "http://central-application-gateway.kyma-system:8080/proxy-cases/code-203"
+    - displayName: code 500
```
Comment (mvshao) [inline tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/proxy-cases.yml:39]: Could you consider adding this test?

https://github.com/kyma-project/kyma/blob/7ff0070341f136ea885d8c66efec4cc896fac3ea/tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/proxy-errors.yml#L12-L21

It was removed due to incompatibility of Golang and http package when proxying 1xx requests. If the problems persist you can hold off on adding it.

File: components/central-application-gateway/internal/externalapi/externalapi.go [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
 
 	"github.com/gorilla/mux"
 )
 
-func NewHandler() http.Handler {
+func NewHandler(lvl zap.AtomicLevel) http.Handler {
 	router := mux.NewRouter()
 
 	router.Path("/v1/health").Handler(NewHealthCheckHandler()).Methods(http.MethodGet)
+	router.Path("/v1/loglevel").Handler(lvl)
```
[the remark this answers is not in the document]
Comment (VOID404) [reply components/central-application-gateway/internal/externalapi/externalapi.go:14]: `Method()` adds routing based on method. Without `Method()` all methods will be passed to this handler. I think I could limit it to `GET` and `PUT`

File: components/central-application-gateway/pkg/httptools/logging.go (hunk 1 above)
Comment (VOID404) [reply components/central-application-gateway/pkg/httptools/logging.go:65]: I am not sure what you mean? `LogRequest` already logs `req.URL.RequestURI()`

Comment (mvshao) [reply components/central-application-gateway/pkg/httptools/logging.go:65]: The `fullPath` variable is not used in this function

Comment (VOID404) [reply components/central-application-gateway/pkg/httptools/logging.go:61]: isn't `fullPath` equivalent to `r.URL.RequestURI()`?

**Review by mvshao — APPROVED** [review]

File: components/central-application-gateway/README.md [hunk 5]
```diff
@@ -38,15 +38,17 @@ To start Central Application Gateway, run this command:
...
-- **proxyPortCompass** is the port that acts as a proxy for the calls from services and Functions to an external solution in the Compass mode. The default port is `8082`.
-- **externalAPIPort** is the port that exposes the API which allows checking the component status. The default port is `8081`.
-- **applicationSecretsNamespace** is the Namespace in which the Application secrets used by Central Application Gateway exist. The default Namespace is `kyma-system`.
-- **requestTimeout** is the timeout for requests sent through Central Application Gateway, expressed in seconds. The default value is `1`.
-- **requestLogging** is the flag for logging incoming requests. The default value is `false`.
-- **proxyTimeout** is the timeout for requests sent through the proxy, expressed in seconds. The default value is `10`.
-- **proxyCacheTTL** is the time to live of the remote API information stored in the proxy cache, expressed in seconds. The default value is `120`.
 
+- **apiServerURL** - The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.
```
Comment (grego952) [inline components/central-application-gateway/README.md:42]:
```suggestion
- **apiServerURL** - The address of the Kubernetes API server. Overrides any value in a kubeconfig. Only required if out-of-cluster.
```

Comment (grego952) [inline components/central-application-gateway/README.md:42]:
```suggestion
- **apiServerURL** - The address of the Kubernetes API server. Overrides any value in a kubeconfig. Only required if out-of-cluster.
```

File: components/central-application-gateway/README.md [hunk 6]
```diff
@@ -38,15 +38,17 @@ To start Central Application Gateway, run this command:
...
+- **applicationSecretsNamespace** - Namespace where Application secrets used by the Application Gateway exist. The default is `kymasystem`
+- **externalAPIPort** - Port that exposes the API which allows checking the component status and exposes log configuration. The default is `8081`
+- **kubeConfig** - Path to a kubeconfig. Only required if out-of-cluster
+- **logLevel** - Log level: panic | fatal | error | warn | info | debug. Can't be lower than info. The default is  zapInfoLevel
```
Comment (grego952) [inline components/central-application-gateway/README.md:46]:
```suggestion
- **logLevel** - Log level: `panic` | `fatal` | `error` | `warn` | `info` | `debug`. Can't be lower than `info`. The default is  `zapInfoLevel`
```

**Review by mvshao — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**